thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java

```diff
@@ -123,6 +137,49 @@ public PinotThirdEyeResponse execute(ThirdEyeRequest request) throws Exception {
     return resp;
   }
 
+  /**
+   * Definition of Pre-Computed Data: the data that has been pre-calculated or pre-aggregated, and does not require
+   * further aggregation (i.e., aggregation function of Pinot should do no-op). For such data, we assume that there
+   * exists a dimension value named "all", which is user-definable keyword in collection configuration, that stores
+   * the pre-aggregated value.
+   * By default, when a query does not specify any value on a certain dimension, Pinot aggregates all values at that
+   * dimension, which is an undesirable behavior for pre-computed data. Therefore, this method modifies the request's
+   * dimension filters such that the filter could pick out the "all" value for that dimension.
+   * Example: Suppose that we have a dataset with 3 dimensions: country, pageName, and osName, and the pre-aggregated
+   * keyword is 'all'. Further assume that the original request's filter = {'country'='US, IN'} and GroupBy dimension =
+   * pageName, then the decorated request has the new filter = {'country'='US, IN', 'osName' = 'all'}.
+   *
+   * @param filterSet the original filterSet, which will NOT be modified.
+   *
+   * @return a decorated filter set for the given precomputed dataset.
+   */
+  public static Multimap<String, String> generateDecoratedFilterSetForPrecomputedDataset(
```

Comment:
Maybe it's just me, but "decorate filter set" doesn't really convey that we are putting in pre aggregate value. Since you're touching this part, should we rename it? What do you think?